protocols/satisfactory.js

```diff
@@ -0,0 +1,92 @@
+import Core from './core.js'
+
+/**
+ * Satisfactory servers run by default self-signed certificates. This is used to ignore the validation for the requests.
+ */
+const https = {
+  rejectUnauthorized: false
+}
+
+export default class satisfactory extends Core {
+  constructor () {
+    super()
+
+    /**
+     * To get information about the Satisfactory game server, you need to first obtain a client authenticationToken.
+     * https://satisfactory.wiki.gg/wiki/Dedicated_servers/HTTPS_API
+     */
+
+    this.authenticationToken = null
+
+    // Don't use the tcp ping probing
+    this.usedTcp = true
+
+  }
+
+  async run (state) {
+    await this.getClientAuthenticationToken()
+
+    await this.queryInfo(state)
+    await this.cleanup(state)
+  }
+
+  async getClientAuthenticationToken () {
+    this.logger.debug('Requesting client access token ...')
+
+    const url = `https://${this.options.host}:${this.options.port}/api/v1/`
+
+    const json = {
+      function: 'PasswordlessLogin',
+      data: {
+        MinimumPrivilegeLevel: 'Client'
+      }
+    }
+
+    const headers = {
+      'Content-Type': 'application/json'
+    }
+
+    this.logger.debug(`POST: ${url}`)
+    const response = await this.request({ url, json, https, headers, method: 'POST', responseType: 'json' })
```

Comment:
The difference between this chunk and the one for `queryInfo` is just the json and headers, we can make a factory function for it to avoid code duplication.

Reply:
Love it!